pyo3-build-config/src/impl_.rs

```diff
@@ -1685,6 +1685,8 @@ fn get_env_interpreter() -> Option<PathBuf> {
 ///   4. `python3`, as above
 pub fn find_interpreter() -> Result<PathBuf> {
     if let Some(exe) = env_var("PYO3_PYTHON") {
+        println!("cargo:rerun-if-env-changed=PYO3_PYTHON_VERSION");
+        println!("cargo:rerun-if-env-changed=PYO3_PYTHON_IMPLEMENTATION");
```

Comment:
Though we need to decide whether these two env vars should override version and implementation we get from the Python interpreter like what `PYO3_CROSS_PYTHON_VERSION` and `PYO3_CROSS_PYTHON_IMPLEMENTATION`. I prefer not to do that, what do you guys think?

Reply:
Hhhmmm, I think if these are passed explicitly, they should override what the interpreter says. We could issue a warning if their contents does not match what the interpreter says though.

Reply:
I prefer not to because they are not intended to be used to override Python version and implementation. Perhaps it's better to be named as `_PYO3_XXX` if we agree to not override.

That said, I'm not strongly opposed to make them override what the interpreter says.

Reply:
If we really want them only to trigger rebuilds, how about `PYO3_ENVIRONMENT_SIGNATURE` and document that this has no other effect than to trigger rebuilds if its value changes. It would then be up to the other layers of automation like Maturin what they put in there (like interpreter version used by the venv).

Reply:
An opaque variable is certainly better, thanks for the idea!